Realm/RLMRealm.mm

```diff
@@ -229,29 +241,25 @@ static void waitForPartialSyncSubscriptions(Realm::Config const& config) {
     row.set(expires_at_col, Timestamp(0, 0));
     realm->commit_transaction();
 
-    NotificationToken token;
-    Results results(realm, table);
-    CFRunLoopRef runLoop = CFRunLoopGetCurrent();
-    CFRunLoopPerformBlock(runLoop, kCFRunLoopDefaultMode, [&]() mutable {
-        token = results.add_notification_callback([&](CollectionChangeSet const&, std::exception_ptr) mutable {
-            if (table->size() > 1) {
-                token = {};
-                CFRunLoopStop(runLoop);
-            }
-        });
+    struct State {
```

Comment:
[q] Assumedly this struct exists to maintain a reference to the `NotificationToken` during the lifecycle of the results notification callback?

Reply:
Yes. The previous version of this code blocked the calling thread until it completed and so could stick everything on the stack, but this version doesn't so it needs to store everything needed for the callback on the heap.